### PR TITLE
Adding requestV2.thrift

### PR DESCRIPTION
**Pull request by kishoreg**
The first part of https://github.com/winedepot/pinot/issues/14

Adding query.thrift which defines the new PinotQuery struct. BrokerRequest has PinotQuery which is optional for now.

### CODE REVIEW

**Review by xiangfu0 — APPROVED** [review]

**Comment by mayankshriv** [comment]
Since you are open, I think PinotRequest is probably a better name than BrokerRequestV2, since v2 containing v1 is a bit odd/confusing.

**Review by mayankshriv — APPROVED** [review]
Other than the naming suggestion, Lgtm.

**Comment by kishoreg** [comment]
Changed it to PinotQuery. BrokerRequest can now wrap the PinotQuery object.

File: pinot-common/src/thrift/query.thrift [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  1: optional string tableName;
+}
+
+struct PinotQuery {
+  1: optional i32 version;
+  2: optional DataSource dataSource;
+  3: optional list<Expression> selectList;
+  4: optional Expression filterExpression;
+  5: optional list<Expression> groupByList;
+  6: optional list<Expression> orderByList;
```
Comment (agrawaldevesh) [inline pinot-common/src/thrift/query.thrift:31]: I think you want a List<Pair<Expression, OrderingHint>> here instead. Where OrderingHint = ASC|DESC

File: pinot-common/src/thrift/query.thrift [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  3: optional Literal literal;
+  4: optional Identifier identifier;
+}
+
+struct Identifier {
+  1: required string name;
+}
+
+struct Literal {
+  1: required string value;
```
Comment (agrawaldevesh) [inline pinot-common/src/thrift/query.thrift:64]: How do we represent non string literals like int or bool ? Should we have a 'Type' field in the Literal too ?

File: pinot-common/src/thrift/query.thrift [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+struct InstanceRequestV2 {
+  1: required i64 requestId;
+  3: optional list<string> searchSegments;
+  4: optional bool enableTrace;
+  5: optional string brokerId;
+}
+
+enum ExpressionType {
```
Comment (agrawaldevesh) [inline pinot-common/src/thrift/query.thrift:46]: Will logical and of expressions be represented using a function ? OR(x, y) ?

File: pinot-common/src/thrift/query.thrift [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+namespace java org.apache.pinot.common.request
+
+struct DataSource {
+  1: optional string tableName;
+}
+
+struct PinotQuery {
+  1: optional i32 version;
+  2: optional DataSource dataSource;
+  3: optional list<Expression> selectList;
```
Comment (agrawaldevesh) [inline pinot-common/src/thrift/query.thrift:28]: A question on semantics: Will the groupByList now need to be contained in the selectList ?

I like the fact that we are not representing aggregation and non-aggregation functions separately. That would help us retain the order.

File: pinot-common/src/thrift/query.thrift (hunk 1 above)
Comment (kishoreg) [reply pinot-common/src/thrift/query.thrift:31]: we can represent that as a function. ASC(Expression)

File: pinot-common/src/thrift/query.thrift (hunk 4 above)
Comment (agrawaldevesh) [inline pinot-common/src/thrift/query.thrift:28]: What about alias's ? 

for example 'select a as one, b as two ...'

File: pinot-common/src/thrift/query.thrift [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+struct InstanceRequestV2 {
+  1: required i64 requestId;
+  3: optional list<string> searchSegments;
+  4: optional bool enableTrace;
+  5: optional string brokerId;
+}
+
+enum ExpressionType {
+  LITERAL,
+  IDENTIFIER,
```
Comment (agrawaldevesh) [inline pinot-common/src/thrift/query.thrift:48]: Will quotes (backticks) be handled by "LITERAL" ?

**Review by agrawaldevesh — COMMENTED** [review]
Sorry for being late to the party !

File: pinot-common/src/thrift/query.thrift (hunk 2 above)
Comment (kishoreg) [reply pinot-common/src/thrift/query.thrift:64]: yes, we can add that. not sure if the parser supports them for now.

File: pinot-common/src/thrift/query.thrift (hunk 3 above)
Comment (kishoreg) [reply pinot-common/src/thrift/query.thrift:46]: yes

File: pinot-common/src/thrift/query.thrift (hunk 4 above)
Comment (kishoreg) [reply pinot-common/src/thrift/query.thrift:28]: It's not a must right now but we can enforce that.

Comment (kishoreg) [reply pinot-common/src/thrift/query.thrift:28]: thats also a function AS (expression)

File: pinot-common/src/thrift/query.thrift (hunk 5 above)
Comment (kishoreg) [reply pinot-common/src/thrift/query.thrift:48]: It should be. We are still working on the parser.

**Pull request merged**